src/test/java/org/jabref/gui/search/GlobalSearchBarTest.java

```diff
@@ -0,0 +1,128 @@
+import org.testfx.framework.junit5.Start;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@GUITest
+@ExtendWith(ApplicationExtension.class)
+public class GlobalSearchBarTest {
+    private Stage stage;
+    private Scene scene;
+    private HBox hBox;
+
+    private GlobalSearchBar searchBar;
+    private StateManager stateManager;
+
+    @Start
+    public void onStart(Stage stage) {
+        SearchPreferences searchPreferences = mock(SearchPreferences.class);
+        when(searchPreferences.getSearchFlags()).thenReturn(EnumSet.noneOf(SearchRules.SearchFlags.class));
+        PreferencesService prefs = mock(PreferencesService.class, Answers.RETURNS_DEEP_STUBS);
+        when(prefs.getSearchPreferences()).thenReturn(searchPreferences);
+
+        stateManager = new StateManager();
+        // need for active database, otherwise the searchField will be disabled
+        stateManager.setActiveDatabase(new BibDatabaseContext());
+
+        // instantiate GlobalSearchBar class, so the change listener is registered
+        searchBar = new GlobalSearchBar(
+                mock(JabRefFrame.class),
+                stateManager,
+                prefs,
+                mock(CountingUndoManager.class),
+                mock(DialogService.class)
+        );
+
+        hBox = new HBox(searchBar);
+
+        scene = new Scene(hBox, 400, 400);
+        this.stage = stage;
+        stage.setScene(scene);
+
+        stage.show();
+    }
+
+    /**
+     * The recordingSearchQueriesOnFocusLost method, accepts a robot reference
+     * to the FxRobot class and verifies that unnecessary prefixes of the
+     * search query are not recorded. Essentially, the query is recorded
+     * only when the searchField node loses focus.
+     *
+     * @param robot Reference to the FxRobot class, a TestFx utility class
+     */
+    @Test
+    void recordingSearchQueriesOnFocusLost(FxRobot robot) throws InterruptedException {
+        stateManager.clearSearchHistory();
+        String searchQuery = "Smith";
+        // track the node, that the search query will be typed into
+        TextInputControl searchField = robot.lookup("#searchField").queryTextInputControl();
+
+        // the focus is on searchField node, as we click on the search box
+        var searchFieldRoboto = robot.clickOn(searchField);
+        for (char c : searchQuery.toCharArray()) {
+            searchFieldRoboto.write(String.valueOf(c));
+            Thread.sleep(401);
+            Assertions.assertTrue(stateManager.getWholeSearchHistory().isEmpty());
+        }
+
+        /*
+         * Set the focus to another node to trigger the listener and finally
+         * record the query.
+         */
+        DefaultTaskExecutor.runInJavaFXThread(() -> hBox.requestFocus());
+        List<String> lastSearchHistory = stateManager.getWholeSearchHistory();
+        List<String> expected = List.of("Smith");
+
+        Assertions.assertEquals(expected, lastSearchHistory);
```

Comment:
See the other tests - please do a static import of `assertEquals`

Reply:
This test fails here:

```
org.opentest4j.AssertionFailedError: expected: java.util.ImmutableCollections$List12@3bfad369<[Smith]> but was: com.sun.javafx.collections.ObservableListWrapper@25e85447<[Smith]>
Expected :[Smith]
Actual   :[Smith]
```

I assume, you have to do:

    assertEquals(FXCollections.observableArrayList("Smith"), lastSearchHistory);

(No need to declare the expected variable - it is less than 20 characters code in your case. And in my suggestion, it it sstill readable).